test/TestCase.cpp

```diff
@@ -149,17 +149,26 @@ void EVMVersionRestrictedTestCase::processBytecodeFormatSetting()
 	// EOF only available since Osaka
 	solAssert(!eofVersion.has_value() || solidity::test::CommonOptions::get().evmVersion().supportsEOF());
 
-	std::string bytecodeFormatString = m_reader.stringSetting("bytecodeFormat", "legacy,>=EOFv1");
-	if (bytecodeFormatString == "legacy,>=EOFv1" || bytecodeFormatString == ">=EOFv1,legacy")
+	auto const bytecodeFormatStr = m_reader.stringSetting("bytecodeFormat", "legacy,>=EOFv1");
+	if (bytecodeFormatStr == "legacy,>=EOFv1" || bytecodeFormatStr == ">=EOFv1,legacy")
+		m_bytecodeFormat = {BytecodeFormat::Legacy, BytecodeFormat::EOFv1};
+	else if (bytecodeFormatStr == "legacy")
+		m_bytecodeFormat = {BytecodeFormat::Legacy};
+	else if (bytecodeFormatStr == ">=EOFv1")
+		m_bytecodeFormat = {BytecodeFormat::EOFv1};
+	else if (bytecodeFormatStr.empty())
+		BOOST_THROW_EXCEPTION(std::runtime_error{"The 'bytecodeFormat' setting requires at least one value."});
+	else
+		BOOST_THROW_EXCEPTION(std::runtime_error{"Invalid bytecodeFormat flag: \"" + bytecodeFormatStr + "\""});
+
+	if (bytecodeFormat().contains(BytecodeFormat::Legacy) && bytecodeFormat().contains(BytecodeFormat::EOFv1))
 		return;
 
 	// TODO: This is naive implementation because for now we support only one EOF version.
-	if (bytecodeFormatString == "legacy" && eofVersion.has_value())
+	if (!bytecodeFormat().contains(BytecodeFormat::EOFv1) && eofVersion.has_value())
 		m_shouldRun = false;
-	else if (bytecodeFormatString == ">=EOFv1" && !eofVersion.has_value())
+	else if (bytecodeFormat().contains(BytecodeFormat::EOFv1) && !eofVersion.has_value())
```

Comment:
To clarify, this is what I meant by "same for the `legacy` run" in https://github.com/ethereum/solidity/pull/15819#discussion_r1994529155:

```suggestion
	else if (!bytecodeFormat().contains(BytecodeFormat::Legacy) && !eofVersion.has_value())
```